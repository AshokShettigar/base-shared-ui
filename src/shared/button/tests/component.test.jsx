import { shallow } from 'enzyme';

import Button from '../component';

describe('Button component', () => {
    const requiredProps = {
        children: 'children',
        onClick: jest.fn()
    };

    it('should render component with default props', () => {
        const wrapper = shallow(<Button {...requiredProps} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with kind prop', () => {
        const wrapper = shallow(<Button {...requiredProps} kind="secondary" />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with icon prop', () => {
        const wrapper = shallow(<Button {...requiredProps} icon="calendar" />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with class size prop', () => {
        const wrapper = shallow(<Button {...requiredProps} size="tiny" />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with loading prop', () => {
        const wrapper = shallow(<Button {...requiredProps} loading />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with active prop', () => {
        const wrapper = shallow(<Button {...requiredProps} active />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with disabled prop', () => {
        const wrapper = shallow(<Button {...requiredProps} disabled />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with className prop', () => {
        const wrapper = shallow(<Button {...requiredProps} className="className" />);

        expect(wrapper).toMatchSnapshot();
    });

    it('shouldn\'t render component', () => {
        const wrapper = shallow(<Button {...requiredProps} rendered={false} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render hidden component', () => {
        const wrapper = shallow(<Button {...requiredProps} hidden />);

        expect(wrapper).toMatchSnapshot();
    });
});
