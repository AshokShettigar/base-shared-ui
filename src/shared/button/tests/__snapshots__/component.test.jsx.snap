// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render component with active prop 1`] = `
<Button
  active={true}
  disabled={false}
  hidden={false}
  kind="primary"
  loading={false}
  onClick={[Function]}
  size="small"
  type="submit"
>
  children
</Button>
`;

exports[`Button component should render component with class size prop 1`] = `
<Button
  active={false}
  disabled={false}
  hidden={false}
  kind="primary"
  loading={false}
  onClick={[Function]}
  size="tiny"
  type="submit"
>
  children
</Button>
`;

exports[`Button component should render component with className prop 1`] = `
<Button
  active={false}
  className="className"
  disabled={false}
  hidden={false}
  kind="primary"
  loading={false}
  onClick={[Function]}
  size="small"
  type="submit"
>
  children
</Button>
`;

exports[`Button component should render component with default props 1`] = `
<Button
  active={false}
  disabled={false}
  hidden={false}
  kind="primary"
  loading={false}
  onClick={[Function]}
  size="small"
  type="submit"
>
  children
</Button>
`;

exports[`Button component should render component with disabled prop 1`] = `
<Button
  active={false}
  disabled={true}
  hidden={false}
  kind="primary"
  loading={false}
  onClick={[Function]}
  size="small"
  type="submit"
>
  children
</Button>
`;

exports[`Button component should render component with icon prop 1`] = `
<Button
  active={false}
  disabled={false}
  hidden={false}
  icon="calendar"
  kind="primary"
  loading={false}
  onClick={[Function]}
  size="small"
  type="submit"
>
  children
</Button>
`;

exports[`Button component should render component with kind prop 1`] = `
<Button
  active={false}
  disabled={false}
  hidden={false}
  kind="secondary"
  loading={false}
  onClick={[Function]}
  size="small"
  type="submit"
>
  children
</Button>
`;

exports[`Button component should render component with loading prop 1`] = `
<Button
  active={false}
  disabled={false}
  hidden={false}
  kind="primary"
  loading={true}
  onClick={[Function]}
  size="small"
  type="submit"
>
  children
</Button>
`;

exports[`Button component should render hidden component 1`] = `
<div
  hidden={true}
>
  <Button
    active={false}
    disabled={false}
    hidden={true}
    kind="primary"
    loading={false}
    onClick={[Function]}
    size="small"
    type="submit"
  >
    children
  </Button>
</div>
`;

exports[`Button component shouldn't render component 1`] = `null`;
